src/google-classroom-extractor/google_classroom_extractor/api/students.py

```diff
@@ -0,0 +1,45 @@
+# SPDX-License-Identifier: Apache-2.0
+# Licensed to the Ed-Fi Alliance under one or more agreements.
+# The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+# See the LICENSE and NOTICES files in the project root for more information.
+
+import logging
+from typing import List, Dict
+import pandas as pd
+from .api_caller import call_api
+
+
+def request_students(resource, course_id: str) -> List[Dict[str, str]]:
+    return call_api(
+        resource.courses().students().list,
+        {"courseId": course_id},  # type: ignore
+        "students",
+    )
+
+
+def request_latest_students_as_df(resource, course_ids: List[str]) -> pd.DataFrame:
+    logging.info("Pulling student data")
+    students: List[Dict[str, str]] = []
+    for course_id in course_ids:
+        students.extend(request_students(resource, course_id))
+
+    return pd.json_normalize(students).astype("string")
+
+
+def request_all_students_as_df(resource, course_ids: List[str], sync_db) -> pd.DataFrame:
```

Comment:
Please consider adding docstrings

Reply:
Can do

Reply:
Going to add this to FIZZ-77.  Lots of refactoring to happen then.